language/stdlib/modules/libra_system.mvir

```diff
@@ -187,8 +187,9 @@ module LibraSystem {
         block_metadata_ref = borrow_global_mut<BlockMetadata>(0xA550C18);
 
         // TODO: Figure out a story for errors in the system transactions.
-        assert(copy(timestamp) > *(&copy(block_metadata_ref).timestamp), 5001);
-        assert(Self.is_validator(copy(proposer)), 5002);
+        // NIL blocks have the same timestamp as their parent and don't have real proposer so use 0x0 as placeholder
```

Comment:
To strongly distinguish between NIL and normal blocks, it might make sense to do something like:

```
if (copy(proposer) == 0x0) {
   // NIL block with null address as proposer. Timestamp must be equal.
   assert(copy(timestamp) >= *(&copy(block_metadata_ref).timestamp), 5001);
} else {
  // Normal block. Time must advance
  assert(copy(timestamp) > *(&copy(block_metadata_ref).timestamp), 5001);	
  assert(Self.is_validator(copy(proposer)), 5002);	
}
```